src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
async_wait_and_get_response or  windowed async_wait are only used for waiting when all useful work is already queued. So i don‘t see why this could be bad for utilization.
For the work submission phase it should stay operation(..., wait=False) and frequent async_response() polling.